python/activator/activator.py

```diff
@@ -164,15 +183,16 @@ def next_visit_handler() -> Tuple[str, int]:
                 expected_visit.detector,
             )
             if oid:
+                m = re.match(oid_regexp, oid)
                 mwi.ingest_image(oid)
-                snap_set.add(snap)
+                expid_set.add(m.group('expid'))
```

Comment:
If by "align" you mean using the right number in the first place, then I think the comment would be more appropriate in `upload.py`.

Reply:
Why not both?

Reply:
Because `activator` has no control over the number, it just knows that that particular part of the string is supposed to be an "exposure ID" (and not even any specific concept of what an "exposure ID" is, which is why the regex accepts almost anything).

Reply:
Fair point. I do wish we weren't doing regex string parsing for this, but we can leave that to KT to clean up.